platform/broadcom/sonic-platform-modules-juniper/qfx5210/modules/x86-64-juniper-qfx5210-64x-sfp.c

```diff
@@ -0,0 +1,1579 @@
+/*
+ * SFP driver for juniper qfx5210_64x sfp
```

Comment:
why not use optoe module?

Reply:
SFPs / QSFPs are managed by a CPLD in QFX5210 platform. So we need a cpld driver to detect the presence of pluggables. The same driver is used for the SFP eeprom mapping to sysfs and the other sfp specific parameters.

AFAIK, optoe driver support only eeprom and port_name in sysfs. Other parameters are CPLD dependent and need to be read from user space.

At this point in time, our preference is to use sfp cpld driver. Please let me know if there are any concerns with this approach.

Reply:
yes, for eeprom and port_name in sysfs please reuse optoe driver. for others like reset, lpmode, presense, it is fine to use cpld driver.

Reply:
We will have to change the OIR logic, validate a number of optics, etc if we move to optoe driver. It will take some good amount of time. Meanwhile we need to get the SONiC workflow in place as Juniper is contributing for the first time. So if the qfx5210 platform support is added to SONiC image, it will immensely help us.

Please allow this initial commit. We will address it in the subsequent commit.

Reply:
I can open an issue for moving to optoe driver. Please let us know if this is acceptable to you for merging the qfx5210 platform patches.

Reply:
@xinliu-seattle  @lguohan Are you fine with this suggestion? Please let me know.

Reply:
@lguohan Waiting for your input.

Reply:
@lguohan As I requested before, can I phase in the optoe changes in the subsequent commit?

Reply:
can you give us an ETA to move optoe?

Reply:
@lguohan ETA will be 9/21. Three weeks from now. Shall try to finish it as early as possible.

Reply:
@lguohan As per our last discussion, could you merge this patch set?